sharding-core/sharding-core-common/src/test/java/org/apache/shardingsphere/core/strategy/keygen/SnowflakeShardingKeyGeneratorTest.java

```diff
@@ -158,6 +167,32 @@ public void assertSetWorkerIdFailureWhenTooMuch() {
         keyGenerator.generateKey();
     }
     
+    @Test(expected = IllegalArgumentException.class)
+    public void assertSetMaxVibrationOffsetFailureWhenTooMuch() {
+        SnowflakeShardingKeyGenerator keyGenerator = new SnowflakeShardingKeyGenerator();
+        Properties properties = new Properties();
+        properties.setProperty("max.vibration.offset", String.valueOf(4096));
+        keyGenerator.setProperties(properties);
+        keyGenerator.generateKey();
+    }
+    
+    @Test
+    @SneakyThrows
+    public void assertSetMaxVibrationOffsetFailureWhenNPowerOf2() {
```

Comment:
I read the test method name seems like I will get exception or set offset failure when the values if power of 2.
But your test case seem to assert set successfully.
So I think the test has some error.